src/arch/ia32e/descriptor/tss.rs

```diff
@@ -1,59 +1,123 @@
-use super::split_address;
-use crate::arch::ia32e::{descriptor::Granularity, PrivilegeLevel};
+//! A module containing the implementation, tests and revolving types around
+//! the [`TssDescriptor`] structure to describe the permissions and capacities
+//! of a task state segment.
 use crate::arch::ia32::descriptor::tss::TssDescriptor as IA32TssDescriptor;
+use crate::arch::ia32e::{descriptor::Granularity, PrivilegeLevel};
+use bit_field::BitField;
 
+/// A task state segment descriptor structure that can be used directly by the
+/// processor to describe a task state segment.
 #[must_use]
-#[derive(Copy, Clone)]
+#[derive(Default, Copy, Clone)]
 #[repr(C, packed)]
 pub struct TssDescriptor {
     tss: IA32TssDescriptor,
+    /// Bits 32 to 63 of the segment base address.
     base_63_32: u32,
+    /// Reserved bits.
     reserved: u32,
 }
 
 impl TssDescriptor {
```

Comment:
No `fmt::display` implementation here too ?

Reply:
The current display format is not the final one, I'm not too sure about what we should do or what we should need. We'll keep the current code and implement proper display trait for those  structure in another PR.